module/VuFind/src/VuFind/Db/Table/LoginToken.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * Table Definition for login_token
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2023.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Db_Table
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFind\Db\Table;
+
+use Laminas\Db\Adapter\Adapter;
+use VuFind\Db\Row\RowGateway;
+use VuFind\Exception\LoginToken as LoginTokenException;
+
+/**
+ * Table Definition for login_token
+ *
+ * @category VuFind
+ * @package  Db_Table
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class LoginToken extends Gateway
+{
+    /**
+     * Constructor
+     *
+     * @param Adapter       $adapter Database adapter
+     * @param PluginManager $tm      Table manager
+     * @param array         $cfg     Laminas configuration
+     * @param RowGateway    $rowObj  Row prototype object (null for default)
+     * @param string        $table   Name of database table to interface with
+     */
+    public function __construct(
+        Adapter $adapter,
+        PluginManager $tm,
+        $cfg,
+        ?RowGateway $rowObj = null,
+        $table = 'login_token'
+    ) {
+        parent::__construct($adapter, $tm, $cfg, $rowObj, $table);
+    }
+
+    /**
+     * Save a token
+     *
+     * @param string $userId    User identifier
+     * @param string $token     Login token
+     * @param string $series    Series the token belongs to
+     * @param string $browser   User browser
+     * @param string $platform  User platform
+     * @param int    $expires   Token lifetime in days
```

Comment:
This comment says expires is token lifetime in days, but lower down, it's compared against the current Unix timestamp to expire a row. Is something inconsistent somewhere, or am I just not reading carefully enough?

Reply:
Comments were wrong. It is supposed to be a timestamp. Comments fixed.